scripts/tasks.py

```diff
@@ -509,3 +511,118 @@ def timetool_correct(config):
             logger.info('No model found! Will return the nominal delay uncorrected!')
 
     tt.timetool_correct(run, nominal, model, figs)
+
+def bayesian_optimization(config):
+    from btx.diagnostics.bayesian_optimization import BayesianOptimization
+    """ Perform an iteration of the Bayesian optimization. """
+    logger.info('Running an iteration of the Bayesian Optimization.')
+    BayesianOptimization.run_bayesian_opt(config)
+    logger.info('Done!')
+    
+def bo_init_samples_configs(config):
+    from btx.diagnostics.bayesian_optimization import BayesianOptimization
+    setup = config.setup
+    task = config.bayesian_opt
+    """ Generates the config files that will be used to generate the initial samples for the Bayesian optimization. """
+    taskdir = os.path.join(setup.root_dir, "bayesian_opt")
+    os.makedirs(taskdir, exist_ok=True)
+    os.makedirs(os.path.join(taskdir, task.tag), exist_ok=True)
+
+    if hasattr(task, 'n_samples_init'):
+        n_samples_init = task.n_samples_init
+
+        # Get the parameters to overwrite
+        params_ranges_keys = [key for key in config if key.startswith("range_")]
+        params_names = [key.replace("range_", "") for key in params_ranges_keys]
+
+        # Generate the parameter samples
+        param_samples = BayesianOptimization.random_param_samples(config)
```

Comment:
This is also not a class function, so can decide to make it one or instantiate the object first.

Reply:
Made it a class method.